deepspeed/inference/engine.py

```diff
@@ -174,6 +179,11 @@ def __init__(self, model, config):
         # Check if local CUDA graphs can be created in replacement modules
         self.local_cuda_graph = self._local_cuda_graph_used(self.module)
 
+    def destroy(self):
+        DeepSpeedTransformerInference.layer_id = 0
+        DeepSpeedSelfAttention.num_layers = 0
+        inference_module.release_workspace()
```

Comment:
Just in case a user wants to explicitly called destroy, might be nice to prevent an `AttributeError` in certain cases

```suggestion
        if inference_module is not None:
            inference_module.release_workspace()
```